### ISSUE TITLE
[Quest] (1249) The Missing Diplomat

### ISSUE THREAD

**Issue by zimms** [unified-db/Database#74]
In this quest, you're supposed to defeat "Slim" who is sneaking out of Menethil's inn.  While Slim does go into stealth to exit the inn, he never becomes attackable, which makes this quest not able to be completed.

Wowhead link: http://www.wowhead.com/quest=1249/the-missing-diplomat

**Comment by xfurry** [comment]
Do you have SD2 installed while testing this? I'm pretty sure this works fine.

**Comment by zimms** [comment]
Yes, I'm pretty sure SD2 was working when this happened, I don't recall anything in the log that indicated otherwise.  But I'm now curious if anyone else can repro.  Oddly, the final quest in the same chain worked, which requires you to "apprehend" a NPC (Reethe) who turns hostile after dialogue (also requiring an SD2 script AFAICT).

**Comment by Grz3s** [comment]
```quote
In this quest, you're supposed to defeat "Slim" who is sneaking out of Menethil's inn. While Slim does go into stealth to exit the inn, he never becomes attackable, which makes this quest not able to be completed.
```
I'll check this soon
Thanks for report

**Comment by Grz3s** [comment]
this quest is done by ScriptDev2
and it must be fixed there.
i'll leave it open untill @xfurry will look at it.

**Comment by Grz3s** [comment]
hey @xfurry 
 I made this q. without SD2...
have a look and say what u think...?
http://paste2.org/IIEhOydx

**Comment by FollowerAI** [comment]
Hello @Grz3s.
Unfortunately, you script is incorrect because, absent evade mechanic.
``` SQL
(1249,1,20,2,0,4962,10873,0x10,0,0,0,0,0,0,0,0,'movement chenged to 2:iwaypoints');
Need replace on
```
``` SQL
(1249,1,32,0,4962,10873,0x10|0x4,0,0,0,0,0,0,0,0,'Set Unpause Waypoint');
```
2) Also need
``` SQL
UPDATE creature SET spawntimesecs = 60, Spawndist = 0, MovementType = 0 WHERE id = 4962; -- movement will be set by q.
```
Replace on
``` SQL
UPDATE creature SET spawntimesecs = 60, Spawndist = 0, MovementType = 2 WHERE id = 4962; -- movement will be set by q.
```
Additional for db_script_on_creature_movement
``` SQL
DELETE FROM dbscripts_on_creature_movement WHERE id BETWEEN 496201 AND 496203;
INSERT INTO dbscripts_on_creature_movement (id, delay, command, datalong, datalong2, buddy_entry, search_radius, data_flags, dataint, dataint2, dataint3, dataint4, x, y, z, o, comments) VALUES
(496201,0,21,1,0,0,0,0x04,0,0,0,0,0,0,0,0,'Tapoke "Slim" Jahn - active'),
(496201,0,21,1,0,0,0,0x04,0,0,0,0,0,0,0,0,'Tapoke "Slim" Jahn - Set Pause Waypoint'),
(496201,1,15,1785,0,0,0,0x04,0,0,0,0,0,0,0,0,'STEALTH'),
(496202,1,22,168,0x01,0,0,0x04,0,0,0,0,0,0,0,0,'temp faction'),
(496203,1,14,1785,0,0,0,0x04,0,0,0,0,0,0,0,0,'STEALTH removed'),
(496203,2,20,0,0,0,0,0x04,0,0,0,0,0,0,0,0,'movement changed to 0:idle'),
(496203,2,21,0,0,0,0,0x04,0,0,0,0,0,0,0,0,'Tapoke "Slim" Jahn - unactive'),
(496203,3,18,0,0,0,0,0x04,0,0,0,0,0,0,0,0,'despawn');
```
To avoid problems with the evade mechanic.

**Comment by Ulduar** [comment]
http://cmangos.net/thread-6849.html
I'll just leave it here. So for the future.

**Comment by Grz3s** [comment]
Hi.. 
Thanks for this report.
Let me defend my mothod.
1st.
Why should i give that npc waypoints and pause them for all servers running time. (so unpause will be only during our q. - so maybe once a day?)...
I decided to ... keep him without any waypoints ..."movementType=0" ... and keeps info about waypints in  "creature_movement_template" -- this info will be used only if someone accepts our quest. This method .. wont make any problems (no errors or logs) ... so why shouldnt be used ??
- Dont u think that closer to blizz will be to give waypoints to someone if he needs them? instead to keep him on 1st wp all the time? 

So in short:
For me command with "pause waypoints" is good if its used during short script/events etc... I'm not a supporter of using it  for creatures that should be "idle" 99% of their time and move only becouse quest etc.

2nd.
```quote
Unfortunately, you script is incorrect because what, absent evade mechanic.
```
I dont get this ... but will try to explain.
-- We dont want him to evade (becouse he will run back to tawern)
As u see ... i used EventAI.. and it work.
- he stops the fight
- he becomes friendly to us
- he does 2 parts of text
- he gives us a q. credit
- he despawns
So all what quest wants..... so whats wrong with this?

3rd.
I appreciate that u want make script work better, but if u doing somethin pls check it after. (i spent on this few hours ... with testing etc and I'm happy how it works - let me ask you - did you test this also? on 3.3.5 server (u know why im asking) ... and then.. did you test your update after that? if yes:
- you making him active before u pause him - so we want him to be active all the time?
- 2nd delay in 03 - you still want him to change movemetn to idle? (even if you changing all to pause?)

Regards

**Comment by Ulduar** [comment]
https://github.com/scriptdev2/scriptdev2-cata/issues/4#issuecomment-49506976
I think he meant this is!
@Schmoozerd answer:
``` QUOTE
this might be a problem with the mmgens being stacked. Hence on evade all but the initial mmgen are remvoved, and if the initial mmgen is not the waypoint-mmgen they will of course return to the spawnpoint.
```
P.S:
The problem is present not only on Cataclysm, but also Wotlk.
In simple language saying,
If the change in the MoveIdle MoveWaypoint, then the cause will not evade Return LastOOC waypoint, and MoveTargetedHome.
You can read:
https://github.com/unified-db/Database/commit/40753afc8cc043dd051e7fa701924ce7a1f8b164#diff-0cc847d380b56fd66cb9222f85af7fb5R108
``` QUOTE
+-- Workaround required because of evade mechanics
```
Not just so it is written!
Therefore, the use EscortAI, there are no such problems even for TemporarySummon Creature.

**Comment by FollowerAI** [comment]
@Grz3s
Excuse me, I reread fully paste2.
Thanks for work. Good example for db_scripts escort quests.
@Ulduar 
Thanks for explanation.

**Comment by Grz3s** [comment]
problem with:
```quote
If the change in the MoveIdle on MoveWaypoint, then the cause will not evade Return LastOOC waypoint, and MoveTargetedHome.
```
is well known .. and for this quest is not important at all. - only chance him to evade.is if he will kill us.
so for death player will be no difference... if he will go back to spawn point or to last reached wp...wont change anything.
Anyway... for this quest. main thing "is to find him" (becouse he is in STEALTH mode)...

**Comment by Ulduar** [comment]
``` QUOTE
duno why u call this q. an escort... we not escorting anyone here / or he not escort us also.
We only need to "damage him" while he tries to run away...
```
@Grz3s https://github.com/scriptdev2/scriptdev2/blob/master/scripts/eastern_kingdoms/wetlands.cpp
https://github.com/scriptdev2/scriptdev2/blob/master/scripts/eastern_kingdoms/wetlands.cpp#L92
                if (Player* pPlayer = GetPlayerForEscort())
https://github.com/scriptdev2/scriptdev2/blob/master/scripts/eastern_kingdoms/wetlands.cpp#L161
            Start(false, (Player*)pInvoker, GetQuestTemplateStore(uiMiscValue), true);
    void UpdateEscortAI(const uint32 uiDiff) override
So he called the escort quest. XD

**Comment by Grz3s** [comment]
ok lets go back to main thing...
until xfurry will say what he thinks about it... i can't accept it or remove it.

**Comment by Ulduar** [comment]
Well, it's there, you decide!
P.S: Although it looks good.
P.P.S: For https://github.com/unified-db/Database/commit/68a75f97dbe8068127f6ccf75c33c113b91247e2
need set respawn coord, Ironaya temporary summon npc.
On Reached Point 5 of course.
I know that this topic does not apply just to alert you before I forget.
Hack solution == On Reached Point 5 cast spell https://github.com/cmangos/mangos-wotlk/commit/160a6a1815b766bd378c39831d642de28662c498
True solution - SD2 script: D.
``` diff
void MovementInform(uint32 uiMoveType, uint32 uiPointId) override
    {
        if (uiMoveType != WAYPOINT_MOTION_TYPE)
            return;

        if (uiPointId == 5)
        // reset home position
        m_creature->SetRespawnCoord(m_creature->GetPositionX(), m_creature->GetPositionY(), m_creature->GetPositionZ(), m_creature->GetOrientation());
    }
```
Note:
The script does not purport to be correct, the goal - to convey information!

**Comment by Ulduar** [comment]
``` QUOTE
EDITED - by Grz3s
-- I appreciate your help - but pls "create new issue - topic if you want to speak about something
-- thank you.
@Grz3s - done.
See https://github.com/unified-db/Database/issues/78

**Comment by xfurry** [comment]
@Grz3s give me some time to review this.
I'm pretty busy this week, so I can't yet comment.

**Comment by Grz3s** [comment]
any luck with this one?

**Comment by xfurry** [comment]
Sorry, I forgot. Will take a look asap.

**Comment by xfurry** [comment]
Line 60 doesn't work. The npc doesn't evade in the right way.
So we can either make the "evade on the spot" work or fix SD2.

**Comment by Grz3s** [comment]
true... i have no any diff option to stop him aggro us...
but this line works -- he stops fight and does what he should do...

And yes ..we seriously need ..smth to drop agrro!

**Comment by xfurry** [comment]
Well I am thinking to have something like ACTION_T_SET_LOCAL_EVADE = true / false
This will enable the npc to evade and stay on the same spot if true or evade and go to home position if false, like it's happening now.

I think it makes sense to have this also for other quests or events. Let me know your thoughts.

**Comment by Grz3s** [comment]
yes.. it would be great.to have it.
another example: (req. same option)
https://github.com/unified-db/Database/issues/88

**Comment by xfurry** [comment]
Well, the function that we are looking for is http://paste2.org/bg495fPs
However I don't think it will work on our case. If the npc evades and gets friendly there won't be any way to complete the quest.

**Comment by xfurry** [comment]
I think this is fixed by https://github.com/scriptdev2/scriptdev2/commit/fcdfa6d5c5ee339323f70cd7df09855fc5c4adbb

**Comment by Grz3s** [comment]
@xfurry 
Yes it does fix ... froblem with faction change and continue his path...
but another problem comes out ..(someone else must confirm) - I tried few Times
After defeating him... he becomes friendly to you...but wont stop attacking you. ..so he cant say his texts and gives u a q. credit..
pls check this part.

**Comment by xfurry** [comment]
Confirmed.
I need to debug that part, to see why it doesn't evade properly.

**Comment by Schmoozerd** [comment]
i think this is related to his guardian pet (still in combat, reaggro, ..)

**Comment by xfurry** [comment]
> i think this is related to his guardian pet (still in combat, reaggro, ..)

No. Pet was killed when he became friendly. And if it's not, it should be despanwed by script before evading.

Somehow he doesn't drop aggro, but I need to see exactly why.

**Comment by Ulduar** [comment]
``` QUOTE
Well, the function that we are looking for is http://paste2.org/bg495fPs
However I don't think it will work on our case. If the npc evades and gets friendly there won't be any way to complete the quest.
```
Why m_IsLocalEvadeEnabled(false)?
Maybe m_bIsLocalEvadeEnabled(false),
Btw - you can get error
/home/travis/build/cmangos/mangos-cata/src/game/CreatureEventAI.cpp:1025:45: error: ‘const struct CreatureEventAI_Action’ has no member named ‘m_IsLocalEvadeEnabled’

**Comment by Grz3s** [comment]
Leave a commenthi @xfurry 
```quote
Well, the function that we are looking for is http://paste2.org/bg495fPs
 However I don't think it will work on our case. If the npc evades and gets friendly there won't be any way to complete the quest.
```
i looked again at your path
In your spare time ... can you pls ... recheck it again... it looks like that all what it does - it forces creature to stop where it is.... but he wont Evade.. like it happens in action 24   ACTION_T_EVADE
We could use it in other quests also.

**Comment by xfurry** [comment]
I will investigate soonish.
I added this to my list.

**Comment by Grz3s** [comment]
Fixed some time ago...
